vllm/v1/sample/rejection_sampler.py

```diff
@@ -288,25 +290,71 @@ def compute_probs(
     # NOTE(woosuk): `apply_top_k_top_p` uses sorting to calculate the mask,
     # which is slow for large vocab sizes. This may cause performance issues.
     logits = apply_top_k_top_p(logits, top_k, top_p)
+
+    if not sampling_metadata.no_penalties:
+        assert sampling_metadata.prompt_token_ids is not None
+
+        _, vocab_size = logits.shape
+
+        # Expand tensors from [batch_size] to [num_tokens].
+        prompt_token_ids = expand_batch_to_tokens(
+            sampling_metadata.prompt_token_ids,
+            cu_num_draft_tokens,
+            num_tokens,
+        )
+        output_tokens_t = _convert_to_tensors(
+            sampling_metadata.output_token_ids, vocab_size, logits.device)
+        output_tokens_t = expand_batch_to_tokens(
+            output_tokens_t,
+            cu_num_draft_tokens,
+            num_tokens,
+        )
```

Comment:
IIUC, we need different "output_token_ids` per draft token position, since the 1st draft token id should be counted for penalties in sampling the 2nd draft token, while the reverse doesn't hold. Is this considered in this PR?